apps/src/aichat/redux/thunks/addChatEvent.ts

```diff
@@ -13,40 +13,46 @@ import {addEventToChatEventsCurrent} from '../slice';
 
 // This thunk adds a chat event to chatEventsCurrent (displayed in current chat workspace) if visible.
 // Then it logs the event to the backend for all chat events except notifications with includeInHistory != true.
+// It also excludes logging to both frontend and backend if the teacher is viewing as a student.
 export const addChatEvent =
   <T extends ChatEvent>(chatEvent: T) =>
   (dispatch: AppDispatch, getState: () => RootState) => {
-    // User action events are hidden from participants and only displayed in teacher view of student chat history.
+    // Do not log to frontend for UserActionEvents (eg, clear chat),
+    // which are hidden from participants and only displayed in teacher view of student chat history.
     if (!isUserActionEvent(chatEvent)) {
       dispatch(addEventToChatEventsCurrent(chatEvent));
     }
-    // Log chat event to backend.
+
+    // Do not log to backend if a teacher is viewing a student's work,
+    // and for most Notifications (exception being when a student fully reset's their project).
+    if (
+      getState().progress.viewAsUserId ||
```

Comment:
The only substantive change here is adding this exception (ie, don't log to server if we're viewing as a student). The logic gating whether to log to the server got pretty confusing though when combined with the notification exception, so I added this early return instead. LMK what you think.